### PR TITLE
chore: add test to macvlan controller

### PR DESCRIPTION
**Pull request by rollandf**
Add env-test for macvlan network controller.

### CODE REVIEW

File: controllers/macvlannetwork_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			verifyNADCreated(cr.GetName(), testNetworkNamespace)
+
+			By("Verify network namespace annotation")
+			verifyNetworkNamespaceAnnot(testNetworkNamespace)
+
+			By("Verify MacvlanNetwork Status")
+			verifyMacvlanStatus(testNetworkNamespace)
+
+			By("Update NetworkNamespace")
+			Eventually(func() error {
```
Comment (ykulazhenkov) [inline controllers/macvlannetwork_controller_test.go:80]: Will this work?

I think it is required to use Gomega instance for Expect calls inside the Eventually block

```
	Eventually(func(g Gomega) {
		model, err := client.Find(1138)
		g.Expect(err).NotTo(HaveOccurred())
		g.Expect(model.Reticulate()).To(Succeed())
		g.Expect(model.IsReticulated()).To(BeTrue())
		g.Expect(model.Save()).To(Succeed())
	}).Should(Succeed())
```

Comment (rollandf) [reply controllers/macvlannetwork_controller_test.go]: Updated.

**Review by ykulazhenkov — APPROVED** [review]
Thx, LGTM

**Review by e0ne — APPROVED** [review]

**Pull request merged**